### PR TITLE
rename slimeball tag, melee tools adjustment, and fix entries for some biome and food tags

### LINKED ISSUE #1373: Melon slices are tagged as "vegetable", rather than "fruit" or "berry"
**Minecraft Version:** 1.21

**NeoForge Version:** 21.0.133-beta

**Logs:** Not relevant.

**Steps to Reproduce:** Not relevant.

**Description of issue:**
I'm currently standardizing Farmer's Delight's usage of `c` tags, and I noticed that `c:foods/vegetable` includes `melon_slice` in it. I find this to be a very odd choice, as melons are botanically and culinarily considered fruits. You wouldn't use a watermelon where you'd use a vegetable when cooking food.

I was instructed by @TelepathicGrunt to open this issue, as the `foods` tag group is meant to be used in food-related features, and this may have been a wrong tag application.

### PR DESCRIPTION
**Pull request by TelepathicGrunt**
Hold on merging until I give approval

Last PR was merged accidentally too early: https://github.com/neoforged/NeoForge/pull/1348

There was still more discussions to be done. Changes here are:

- Added Old Growth Spruce Taiga, Windswept Hills, Windswept Gravelly Hills, Windswept Forest, and Stony Shores to the `c:is_cold/overworld` tag.

- Moved Melon Slice from `c:foods/vegetable` to `c:foods/fruit` as melons are usually considered fruits.

- Removed Glistering Melon Slice from `c:foods/golden` because it is not actually edible and thus, not food.

- Melee weapon tag no longer has pickaxes, hoes, or shovels in it. And clarified the melee weapon tag's javadoc to make it clear pickaxes are excluded because being a weapon is not a pickaxe's primary purpose. 

- rename `c:slimeballs` to `c:slime_balls` and deprecate the older tag. The tag name is closer match to vanilla slime ball registry name. The translation name would have no space to match vanilla slimeball translation name

### CODE REVIEW

**Comment by TelepathicGrunt** [comment]
Fabric side reach agreement and was merged. Merging this now

**Comment by TelepathicGrunt** [comment]
https://github.com/FabricMC/fabric/pull/3957

File: src/main/java/net/neoforged/neoforge/common/data/internal/NeoForgeItemTagsProvider.java [hunk 1]
```diff
@@ -101,7 +101,7 @@ public void addTags(HolderLookup.Provider lookupProvider) {
         copy(Tags.Blocks.FENCES_NETHER_BRICK, Tags.Items.FENCES_NETHER_BRICK);
         copy(Tags.Blocks.FENCES_WOODEN, Tags.Items.FENCES_WOODEN);
         tag(Tags.Items.FERTILIZERS).add(Items.BONE_MEAL);
-        tag(Tags.Items.FOODS_FRUIT).add(Items.APPLE, Items.GOLDEN_APPLE, Items.ENCHANTED_GOLDEN_APPLE);
+        tag(Tags.Items.FOODS_FRUIT).add(Items.APPLE, Items.GOLDEN_APPLE, Items.ENCHANTED_GOLDEN_APPLE).add(Items.MELON);
```
Comment (sciwhiz12) [inline src/main/java/net/neoforged/neoforge/common/data/internal/NeoForgeItemTagsProvider.java]: This is adding the item version of the Melon block, but the Fabric PR adds the melon _slice_ item. See https://github.com/FabricMC/fabric/pull/3957/files#diff-bd8d1f7f942977cc89b7d65685dd0327a87c3656352c817981c821fd142fb269R323.

File: src/main/java/net/neoforged/neoforge/common/data/internal/NeoForgeItemTagsProvider.java [hunk 2]
```diff
@@ -101,7 +101,7 @@ public void addTags(HolderLookup.Provider lookupProvider) {
...
         tag(Tags.Items.FOODS_VEGETABLE).add(Items.CARROT, Items.GOLDEN_CARROT, Items.POTATO, Items.MELON_SLICE, Items.BEETROOT);
```
Comment (sciwhiz12) [inline src/main/java/net/neoforged/neoforge/common/data/internal/NeoForgeItemTagsProvider.java]: This retains the addition of the melon slice item in the vegetables tag, which was removed in the corresponding Fabric PR: https://github.com/FabricMC/fabric/pull/3957/files#diff-bd8d1f7f942977cc89b7d65685dd0327a87c3656352c817981c821fd142fb269L227.

**Review by sciwhiz12 — CHANGES_REQUESTED** [review]

**Review by sciwhiz12 — APPROVED** [review]
Merge at will.

**Pull request merged**